waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -49,6 +58,47 @@ validateExactlyOneAppExists spec =
         GenericValidationError $
           "You have more than one 'app' declaration in your Wasp app. You have " ++ show (length apps) ++ "."
 
+validateWasp :: AppSpec -> [ValidationError]
+validateWasp spec = validateWaspVersion specWaspVersionStr
+  where
+    specWaspVersionStr = Wasp.version $ App.wasp (snd $ getApp spec)
+
+validateWaspVersion :: String -> [ValidationError]
+validateWaspVersion specWaspVersionStr = eitherUnitToErrorList $ do
+  specWaspVersion <- parseWaspVersion specWaspVersionStr
+  unless
+    ( SV.isVersionInRange currentWaspVersion $
+        SV.Range [SV.backwardsCompatibleWith specWaspVersion]
+    )
+    $ Left $ incompatibleVersionError currentWaspVersion specWaspVersion
+  where
+    parseWaspVersion :: String -> Either ValidationError SV.Version
+    parseWaspVersion waspVersionStr = do
+      let (_ :: String, _ :: String, _ :: String, waspVersionDigits :: [String]) =
+            waspVersionStr =~ ("\\^([0-9]+).([0-9]+).([0-9]+)$" :: String)
+      case mapM readMaybe waspVersionDigits of
+        Just [major, minor, patch] -> Right $ SV.Version major minor patch
+        __ -> Left $ GenericValidationError "Wasp version should be in the format ^major.minor.patch"
```

Comment:
Now that we have normal semver, we could actually produce a range here -> because it what we are really dealing with here, with range.

So we could change the signature to
```hs
parseSpecWaspVersionRange :: String -> Either ValidationError -> SV.Range
```

and then also modify `incompatibleVersionError` to be `SV.version -> SV.Range -> ValidationError`.

I think that would make it a bit nicer, more to the point.